foyer/tests/test_forcefield.py

```diff
@@ -74,6 +75,17 @@ def test_write_refs():
     ethane = oplsaa.apply(mol2, references_file='ethane.bib')
     assert os.path.isfile('ethane.bib')
 
+def test_write_refs_multiple():
+    mol2 = mb.load(get_fn('ethane.mol2'))
+    oplsaa = Forcefield(forcefield_files=get_fn('refs-multi.xml'))
+    ethane = oplsaa.apply(mol2, references_file='ethane-multi.bib')
+    assert os.path.isfile('ethane-multi.bib')
+    with open(get_fn('ethane-multi.bib')) as file1:
+        with open('ethane-multi.bib') as file2:
+            diff = difflib.ndiff(file1.readlines(), file2.readlines())
+        changes = [l for l in diff if l.startswith('+ ') or l.startswith('- ')]
```

Comment:
Check out https://docs.python.org/3/library/filecmp.html

Reply:
So I had initially tried using `filecmp` when writing [this test](https://github.com/mosdef-hub/mbuild/blob/1468900d2121661947b221a8f2b5534405ca9fcc/mbuild/tests/test_utils.py#L15-L22) for mBuild; however, I ran into an issue with Appveyor (https://ci.appveyor.com/project/ctk3b/mbuild-o0viu/build/1.0.755#L1137). I think because `filecmp` performs a binary comparison there is an issue that arises when comparing files generated on MacOS/Linux and on Windows (see https://bugs.python.org/issue6306).

Reply:
Ah I see - makes sense. If there is no diff, couldn't you simply `assert not diff`? Or will it always contain something? Is it just a header that you can unambiguously strip?

Regardless of the above, I would de-dent the `changes = ...` and `assert` lines.

Reply:
`diff` will show all lines, with a `+ ` in front of lines that are unique to `file2` and a `- ` in front of lines that are unique to `file1`. So `changes` is just stripping out only those lines that are different between the two files. I actually got this from [this StackOverflow answer](https://stackoverflow.com/questions/15864641/python-difflib-comparing-files/15864920#15864920).

Reply:
Heh alright this isn't really that essential but one last suggestion: https://docs.python.org/2/library/difflib.html#difflib.unified_diff and just set `n=0` to get no context lines! 🎉

Reply:
🍾 Ah, nice suggestion! It seemed like there had to be a better way to compare two files.